dns/resolver.py

```diff
@@ -991,12 +1224,24 @@ def query(self, qname, rdtype=dns.rdatatype.A, rdclass=dns.rdataclass.IN,
                 continue
             all_nxdomain = False
             break
+
         if all_nxdomain:
+            if self.cache:
+                if isinstance(self.cache, LRUPosNegCache):
+                    #response.answer
+                    response.answer = dns.message.QUESTION
+                    answer = Answer(_qname, rdtype, rdclass, response,
+                                    raise_on_no_answer) # DNS reg found
+                    self.cache.put((_qname, rdtype, rdclass), answer)
+                    self.cache.inc_negative_miss()
+                    return answer
+            # Not found (Normal/LRU or no cache is the same use case
             raise NXDOMAIN(qnames=qnames_to_try, responses=nxdomain_responses)
         answer = Answer(_qname, rdtype, rdclass, response,
-                        raise_on_no_answer)
+                        raise_on_no_answer) # DNS reg found
```

Comment:
This is the only line causing a merge conflict.  Back it out, and rebase is easy.